File: server/project/mj-client/src/main/java/majiang/client/portal/admin/form/ValueForm.java
package majiang.client.portal.admin.form;

import org.forkjoin.apikit.core.Message;
import org.hibernate.validator.constraints.NotEmpty;

/**
 *
 */
@Message
public class ValueForm {
    @NotEmpty
    private String value;

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return "ValueForm{" +
                "value='" + value + '\'' +
                '}';
    }
}

File: server/manager/agent-js-sdk/form/ValueForm.js
'use strict';



/**

*/
class ValueForm {

    value:String;
    constructor() {

    }

    formObject({value}):ValueForm{
        this.value = value;
        return this;
    }

    static of({value}):ValueForm{
        return new ValueForm().formObject({value});
    }

    static form(value:String):ValueForm{
        return new ValueForm().formObject({value});
    }
}

export default ValueForm;